### PR TITLE
The account linking user journeys

### PR DESCRIPTION
**Pull request by olatomoloju**
The work involved:
- The 3 account linking user journeys implementation as covered by tickets HAAR 1361, 1362, 1363
- Client side validation on all screens
- Back/Cancel navigation logic implementation on all new screens
- Data persistence across screens
- New user search UI feature, which uses an exposed managed users api endpoint
- Updating the existing 3 non-linking create-dps-user functionality, because the non linking screen flow branches off the  linking-options screen.

### CODE REVIEW

File: backend/controllers/createLinkedDpsUser.test.js [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    createLinkedGeneralUser,
+    searchUser,
+    '/create-linked-dps-user',
+    '/create-dps-user',
+    '/create-user',
+    '/manage-dps-users',
+  )
+
+  describe('index', () => {
+    it('should call create-user index scenario 1', async () => {
```
Comment (AlTurner-MOJ) [inline backend/controllers/createLinkedDpsUser.test.js:23]: What are scenarios 1, 2 & 3? Please include a description of what the scenario under test is i.e.
`it('should call create-user index when <insert scenario description here>', async () => {`

File: backend/controllers/createLinkedDpsUserDecision.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    const errors = validateLinkedUserOptionSelection(user.userExists)
+    // const errors = []
+    if (errors.length > 0) {
+      stashStateAndRedirectToIndex(req, res, errors, [user])
+    } else {
+      // eslint-disable-next-line no-useless-catch
+      try {
+        stashStateAndRedirectToCreateLinkedDpsUser(req, res, [user])
+      } catch (err) {
+        throw err
```
Comment (AlTurner-MOJ) [inline backend/controllers/createLinkedDpsUserDecision.js:55]: Do we need this?

File: backend/controllers/createLinkedDpsUser.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
+const { trimObjValues } = require('../utils/utils')
+const {
+  validateUserForSearch,
```
Comment (ravmoj) [inline backend/controllers/createLinkedDpsUser.js:3]: can we split this file based on search and create functionality

File: backend/controllers/createLinkedDpsUser.test.js [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+      }
+      const render = jest.fn()
+      const redirect = jest.fn()
+      await createLinkedDpsUser.index(req, { render, redirect })
+
+      expect(req.flash).toBeCalledWith('user')
+      expect(redirect).toBeCalledWith('/create-user')
+    })
+
+    it('should call create-user index scenario 3', async () => {
```
Comment (ravmoj) [inline backend/controllers/createLinkedDpsUser.test.js:49]: Please add actual scenario in the test (for above two tests)

File: backend/api/manageUsersApi.js [hunk 5]
```diff
@@ -97,16 +97,24 @@ const manageUsersApiFactory = (client) => {
   const deactivateExternalUser = (context, { userId, reason }) =>
     put(context, `/externalusers/${userId}/disable`, { reason })
```
Comment (ravmoj) [inline backend/api/manageUsersApi.js:99]: branch is out of synch .

File: backend/controllers/createLinkedDpsUser.js (hunk 3 above)
Comment (olatomoloju) [reply backend/controllers/createLinkedDpsUser.js:3]: Yes, I can look into that.

File: backend/controllers/createLinkedDpsUserDecision.js (hunk 2 above)
Comment (olatomoloju) [reply backend/controllers/createLinkedDpsUserDecision.js:55]: Keeping as agreed.

File: backend/controllers/createLinkedDpsUser.test.js (hunk 4 above)
Comment (olatomoloju) [reply backend/controllers/createLinkedDpsUser.test.js]: Done, thanks

File: backend/api/manageUsersApi.js (hunk 5 above)
Comment (olatomoloju) [reply backend/api/manageUsersApi.js:99]: Yes, we can merge changes in main.

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**